### PR TITLE
feat(docs): Describe coordinate systems and reference points (#510)

### LINKED ISSUE #510: Describe coordinate systems and reference points
Create content for the following topics:

- [x] coordinate_systems.adoc
- [x] ~~reference_points.adoc~~

### PR DESCRIPTION
**Pull request by max-rosin**
This PR resolves #510. 

- Added a conceptual topic about coordinate systems and reference points.
- Added a use case topic about transforming messages.

- [x] My suggestion follows the [style and contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html).
- [x] I have taken care about the [documentation](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/commenting.html).
- [x] I have done the [DCO signoff](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html#developer-certification-of-origin-dco).
- [x] My changes generate no errors when passing CI tests. 
- [x] I have successfully implemented and tested my fix/feature locally.
- [x] Appropriate reviewer(s) are assigned.

### CODE REVIEW

**Comment by kmeids** [comment]
Output CCB 12.05.2021:
1. WP Leads to review documentation pull requests set to "ReadyForCCB" by Max.

File: doc/architecture/reference_points_coordinate_systems.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+= Reference points and coordinate systems
+
+With Open Simulation Interface, an object's position coordinates refer to the center of the object's bounding box, independent of the coordinate system. This reference point is not identical to the origin of any coordinate systems.
```
Comment (clemenshabedank) [inline doc/architecture/reference_points_coordinate_systems.adoc:3]: "...independent of the coordinate system." -> can this be dropped? 

I have a hard time to understand the second sentence. Do you mean: "The reference point does not have to be identical to the origin of sensor coordinates or object coordinates"? 

Would it make sense to shift the two sentences under "World coordinates"?

File: doc/architecture/reference_points_coordinate_systems.adoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+World coordinates::
+Coordinate system for all quantities which are part of ground truth.
+
+Sensor coordinates::
+Coordinate system for all quantities which are part of sensor data.
+
+Object coordinates::
+Coordinate system for local object coordinates, for example, vehicle coordinates.
+
```
Comment (clemenshabedank) [inline doc/architecture/reference_points_coordinate_systems.adoc:15]: For all coordinate systems: Would it make sense to say something about the origin? e.g.
World coordinates: inertial x/y/z positions
Sensor coordinates: mounting position of the sensor / virtual mounting position
Object coordinates: For vehicles it is currently middle rare axle, but there were discussions about maybe changing it to middle bounding box

File: doc/usecases/transforming_coordinate_systems.adoc [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+``GroundTruth::moving_object::base::orientation``::
+Defines the orientation of the vehicle’s reference point.
+
+``GroundTruth::moving_object::vehicle_attributes::bbcenter_to_rear``::
+The vector pointing from the vehicle’s reference point to the middle of the rear axle under neutral load conditions in vehicle coordinates.
+The middle of the rear axle is 
+
+``SensorData::mounting_position``::
+Defines the sensor’s position and orientation and thereby the origin of the sensor coordinate system.
+The mounting position is defined in sensor coordinates and relative to the origin of the vehicle coordinate system.
```
Comment (clemenshabedank) [inline doc/usecases/transforming_coordinate_systems.adoc:22]:
```suggestion
The mounting position is given in vehicle coordinates.
```

File: doc/usecases/transforming_coordinate_systems.adoc [hunk 4]
```diff
@@ -0,0 +1,22 @@
+= Transforming coordinate systems
+
+Transforming coordinates for a specific vehicle or sensor is a common task when running simulations.
+OSI uses DIN ISO 8855:2013-11 for coordinate systems and transformations between coordinate systems.
+
+//TODO: Should we add one or more sentences about the mathematical operations involved?
```
Comment (clemenshabedank) [inline doc/usecases/transforming_coordinate_systems.adoc:6]: In my opinion that would be very helpful. Also pictures help here a lot, I think.

File: doc/architecture/reference_points_coordinate_systems.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
+= Reference points and coordinate systems
+
+With Open Simulation Interface, an object's position coordinates refer to the center of the object's bounding box.
```
Comment (kmeids) [inline doc/architecture/reference_points_coordinate_systems.adoc:3]:
```suggestion
In Open Simulation Interface, object's coordinates refer to the position of its center of bounding box.
```

File: doc/architecture/reference_points_coordinate_systems.adoc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+Object coordinate system::
+Coordinate system for local object coordinates, for example, vehicle coordinates.
+For vehicles, the origin is the middle of the rear axle projected onto the ground.
+Other objects may use different origins.
+
+The following image shows an example of the differents coordinate systems.
+It also shows that the reference point of the vehicle is located in the center of its bounding box whereas the origin of the vehicle coordinate system is the middle of rear axle projected onto the ground.
+In this example, the virtual mounting position coincides with the origin of the vehicle coordinate system.
+
+image::{imagedir}/osi_coordinate_systems.png[]
```
Comment (kmeids) [inline doc/architecture/reference_points_coordinate_systems.adoc:26]: We should use a better image for this, one that does not cover the "Z" on the virtual coordinate system and that also shows the virtual coordinate system in a different location from the rear axle to evade confusion with the DIN ISO.

**Review by kmeids — COMMENTED** [review]
Please check suggestions and comments

File: doc/architecture/reference_points_coordinate_systems.adoc [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+Sensor coordinate system::
+Coordinate system for all quantities which are part of sensor data.
+The origin is the mounting position of the physical sensor or a virtual mounting position, depending on the technology.
+
+Object coordinate system::
+Coordinate system for local object coordinates, for example, vehicle coordinates.
+For vehicles, the origin is the middle of the rear axle projected onto the ground.
+Other objects may use different origins.
+
+The following image shows an example of the differents coordinate systems.
```
Comment (kmeids) [inline doc/architecture/reference_points_coordinate_systems.adoc:22]:
```suggestion
The following image shows an example of the coordinate systems.
```

**Review by kmeids — COMMENTED** [review]
Please check suggestions and comments.

File: doc/architecture/reference_points_coordinate_systems.adoc (hunk 2 above)
Comment (kmeids) [reply doc/architecture/reference_points_coordinate_systems.adoc:21]: **Output CCB 09.06.2021:**
1. Describe OSI's Coordinate systems in a generic manner.
2. Describe how to define the position of the coordinate systems in OSI (for example wrt reference points, which may be the center of the bounding box).
3. Example picture to illustrate the above mentioned points (a picture is already available needs to be updated).

File: doc/architecture/reference_points_coordinate_systems.adoc (hunk 6 above)
Comment (max-rosin) [reply doc/architecture/reference_points_coordinate_systems.adoc]: Simplified the image and moved the example to the topic describing the transformations between coordinate systems.
Seemed to me to be the more appropriate place.

**Pull request merged**